syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service BranchService {
  rpc GetAll (GetAllRequest) returns (GetAllBranchesResponse);
  rpc GetById (GetBranchByIdRequest) returns (BranchModel);
}

message BranchModel {
  int32 branch_id = 1;
  optional string name = 2;
  optional string city = 3;
  optional string address = 4;
  optional string phone = 5;
}

message GetAllBranchesResponse {
  repeated BranchModel branches = 1;
}

message GetBranchByIdRequest {
  int32 branch_id = 1;
}
